### DISCUSSION TITLE
Existing feeds don't work and importing these feeds causes error

### DISCUSSION THREAD

**Discussion by kerednekt1a** [rrirower/DP2SceneGraphWizard#64] [General]
Greetings and thank you for your work on this software.
I manually entered the detailof one video and everything worked fine.
But when I pointed the feed to my JSON feed I have used for nearly 2 yearson Roku the main scene was blank.
I pointed the feed to my XML feed and the categories appeared but no posters or descriptions though the videos would play.
I tried importing my xml feed file and then a csv file but both gave an error.
So the record data layout of my JSONand XML and CSV files must be incorrect.
Can you please provide the correct layoutfor me please? 
Manythanks,
Derek

**Comment by ferdiworks** [comment]
@kerednekt1a Please post the URL to your feed so we can look at it in a browser window and determine what may be a problem! Please note there are many feed providers who in the past used feeds without categories or playlists and just used the Direct publisher tag feature which is no longer available so the feeds have to be to the Roku specification with the proper categories and or Queries in the JSON Feed! Note: the main scene was blank. Can be blank for multiple reasons from broken links or code errors or no categories or Quires in the feed! Next, can you explain this,  You wrote that you manually entered the details of one video and everything worked fine? Entered it where? worked fine from where? Please post the feed url!

**Reply by kerednekt1a** [reply]
@ferdiworks 
here is my Json feed 
https://www.tvbossfire.com/static/channels/62cd74c54f1d5d075119b10b.json

Also here is my xml file I use to feed my channel on Amazon Fire TV
https://www.tvbossfire.com/static/channels/62cd74c54f1d5d075119b10b.xml

I used the Channel Feed Manager to create a new feed file and added one shortform video as well as all my categories.
I pointed the feed in the manifest file of the sample template to that new json file.
I zipped sample template and uploaded to my roku device.
The categories were all displayed as was the video poster and description etc. and when I clicked on the poster the video played correctly.

Thank you.

**Reply by ferdiworks** [reply]
@kerednekt1a OK, I am familiar with TV Boss Fire! You have two options! 1) In the backend of TV Boss Fire Craig has a Generic Roku template with video tutorials on how to edit it and the process to create the SDK package and sideload, sign it and, upload it to Roku and what you need to do to submit it for publishing! BUT, the generic template he has will only yield one line of videos for the Shortform videos as his feed from TV Boss Fire does not include Categories and or playlist or Quuieries in the feeds produced by the software! 2) You can edit the feed and add Categories and Playlist so the SDK Package will read the category Playlist or Quireies and output the categories to TV! Op[tion 3) Have someone do it for you! And yes it's not free to get someone to do this and you can incur a charge!  Note I do provide a software solution in the cloud that creates channels but we charge 97 a month but it includes Hosting, Livestreaming and many other functions! If you would like to edit the feed your self you are going to have to add categories and playlist, At the end of your feed you have 3 brackets
                    ]
		}
	],
You then add the code below:
The code looks like this:

`"categories": [
    {
      "name": "All Videos",
     "playlistName":"All",
      "order": "most_popular"
    },
    {
      "name": "Divine Mercy Matters",
      "playlistName":"Divine Mercy Matters",
      "order": "chronological"
    },
    {
      "name": "Discovering the Diary",
      "playlistName":"Discovering the Diary",
      "order": "most_recent"
    },
    {
      "name": "Featured Free for a Limited Time",
      "playlistName":"Featured Free",
      "order": "manual"
    },
    {
      "name": "Explaining the Faith with Fr. Chris Alar",
      "playlistName":"Explaining the Faith",
      "order": "chronological"
    }
  ],
  "playlists": [
    {
      "name": "All",
      "itemIds": [
        "archives-srfaustina1",
        "c8168fc591444f1a8dd57f6066c26c88",
        "01915b3a171a461ebe6e0a098d53e705",
        "2c27a33b4eee4705971b1904490c72be",
        "91d627fdd4dc4dc0a79737cfc5a89428",
        "c9cdab6e098c403a86f776cf13cd68d2",
        "55c605228e584bc6ad3788b754808943",
        "d0f01326bd764b6991b4e8ad165f2c59",
        "39946711a4b5461d9519797a95e09de4",
        "4471a6ab89c444a6ac624fc6ebb69341",
        "0236fd9efc0a4be090b1130526b86880",
        "8cb5b158c6ae403f9ced68af25aabe73",
        "e13adc59658e483fa519f0dca3bcd30a",
        "b2672956cd6044158b2a9187291f6afa",
        "3f5aab9529084f32bd02e901d9ff53b9",
        "235b85bd90844a05865190a0fca659ff"
      ]
    },
    {
      "name": "Divine Mercy Matters",
      "itemIds": [
        "c8c1e4b00b144a649887cccf124522c5"
      ]
    },
    {
      "name": "Discovering the Diary",
      "itemIds": [
        "01915b3a171a461ebe6e0a098d53e705",
        "4471a6ab89c444a6ac624fc6ebb69341",
        "c8168fc591444f1a8dd57f6066c26c88"
      ]
    },
    {
      "name": "Featured Free",
      "itemIds": [
        "f1377ac9543c4b9795239516ef0caa64",
        "38f3697469b4418b82fbb59945ed4aaa"
      ]
    },
    {
      "name": "Explaining the Faith",
      "itemIds": [
        "8cb5b158c6ae403f9ced68af25aabe73",
        "d708716987934c26add6574788ea4117",
        "06e25536ff004058a6453fd86ab1930e",
        "22af8f6370434cf887072b1c7a27a4f8",
        "39946711a4b5461d9519797a95e09de4",
        "e13adc59658e483fa519f0dca3bcd30a",
        "d0f01326bd764b6991b4e8ad165f2c59",
        "2c27a33b4eee4705971b1904490c72be",
        "1c606116194c4f73a32b5d4e6f696fb0",
        "91d627fdd4dc4dc0a79737cfc5a89428",
        "c9cdab6e098c403a86f776cf13cd68d2",
        "55c605228e584bc6ad3788b754808943",
        "0236fd9efc0a4be090b1130526b86880",
        "b2672956cd6044158b2a9187291f6afa",
        "3f5aab9529084f32bd02e901d9ff53b9",
        "84633bc134db49fdbd1075d827581810",
        "e82af69af78a464d8acf8ff0fa6ac775",
        "a0ab3f8ce2854a7a8680ebadfd289452",
        "f3a0024758b148a081df0e2513b270c8",
        "3c2bf497896b4f3eab22d3bf6e239bee",
        "b46997f173374cf2910e0a2b56d68f1c",
        "8322cd8277164eb981f4be57f1438df0"
      ]
    }
  ]
}`

**Reply by ferdiworks** [reply]
The above code is for the JSON Feed

**Reply by rrirower** [reply]
@kerednekt1a I used the link you provided above and downloaded your feed file.  I then tried to validate it with the Channel Feed Manager.  It indicated you have several errors of the type: 

`"Item 'Tennessee' defines tags, but, a category that associates those tags could not be found."`

Without Category objects, the wizard code cannot define rows in the channel.  Thus, nothing will display.

On further inspection, your feed file does not define Category objects and contains field objects that are not defined in the Roku Json Spec.  The wizard software relies on the Roku Json Spec to validate its input.  You may run into further problems in the feed file is not corrected.

**Comment by kerednekt1a** [comment]
@rrirower @ferdiworks 
Thank you both for your feedback - much appreciated.

I had tried the sample template from TVBoss and sought something better.
I had suspected for sometime that the feed was not 100% in order.

I have some coding experience so will try adding the code suggested.
I had tried the Roku sample templates also but seemed always to hit a roadblock.
I assumed the feed was ok as I had been using it for so long.

Many thanks again!

**Comment by kerednekt1a** [comment]
@ferdiworks @rrirower 
Just a quick update. 
I added the categories to the end of my json file, uploaded to my own server .
Directed the feed in the manifest to that file and now my channel works as it should. 
Many thanks for your help.

I just want to add in case anyone else has the same problem that I asked TVBoss if they would add the necessary code for categories etc..
The answer was "Not at this time". 

Again many thanks for your help.
